### PR TITLE
[Azure.Monitor.Query] Update docs to get namespaces

### LINKED ISSUE #30903: MetricsQueryClient returns empty results
[Enter feedback here]

Hello, we tried to use Azure.Monitor.Query package to retrieve metrics data for our resources. We were seeing empty results in MetricResult.TimeSeries in the response, when using below code. No error.

```csharp
string resourceId = "/subscriptions/SUB_ID/resourceGroups/RESOURCE_GROUP/providers/Microsoft.Storage/storageAccounts/STORAGE_ACCOUNT";
MetricsQueryClient metricsClient = new MetricsQueryClient(new DefaultAzureCredential()); 
var blobCount = (decimal)0.0;

var blobCountResult = metricsClient.QueryResource(resourceId, new[] { "BlobCount" },
  new MetricsQueryOptions
  {
      MetricNamespace = "Blob",
      Aggregations =
      {
          MetricAggregationType.Average
      },
      Granularity = TimeSpan.FromMinutes(10),
      TimeRange = new QueryTimeRange(TimeSpan.FromMinutes(60))
  });
```

After troubleshooting multiple times, capturing requests through Fiddler and comparing it with REST API we figured out the values we were passing in the code were not correct.

Correct usage:

```csharp
string resourceId = "subscriptions/SUB_ID/resourceGroups/RESOURCE_GROUP/providers/Microsoft.Storage/storageAccounts/STORAGE_ACCOUNT/blobServices/default";
MetricsQueryClient metricsClient = new MetricsQueryClient(new DefaultAzureCredential()); 
var blobCount = (decimal)0.0;

var blobCountResult = metricsClient.QueryResource(resourceId, new[] { "BlobCount" },
  new MetricsQueryOptions
  {
      MetricNamespace = "Microsoft.Storage/storageAccounts/blobServices",
      Aggregations =
      {
          MetricAggregationType.Average
      },
      Granularity = TimeSpan.FromHours(1),
      TimeRange = new QueryTimeRange(TimeSpan.FromMinutes(60))
  });
```

We had referred to multiple documentation:
[Documentation reference for the package](https://docs.microsoft.com/en-us/dotnet/api/azure.monitor.query?view=azure-dotnet)
[List of supported metrics](https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-supported) & [resource providers](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers)
[Monitoring Storage account ](https://docs.microsoft.com/en-us/azure/storage/blobs/monitor-blob-storage-reference)

Only after we used [Metrics List](https://docs.microsoft.com/en-us/rest/api/monitor/Metrics/List?tabs=HTTP) API and compared it to the request Azure.Monitor.Query package was sending (captured details through fiddler), we were able to understand we were passing the wrong the values.

Queries:
1. Using 'BlobCount' as metric with 'Blob' as namespace should have thrown error. But we did not see any error instead we could see success response with empty result.

2. Can we update the documentation of [Azure.Monitor.Query](https://docs.microsoft.com/en-us/dotnet/api/azure.monitor.query.metricsqueryclient?view=azure-dotnet) package MetricsQueryClient to include some more examples so that we understand what values to use in the options.

Thanks.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9c6e1b7d-47c1-f897-9178-4d32c113bf71
* Version Independent ID: 639e3ecd-deec-d327-f826-03ed36d7efe1
* Content: [MetricsQueryOptions Class (Azure.Monitor.Query) - Azure for .NET Developers](https://docs.microsoft.com/en-us/dotnet/api/azure.monitor.query.metricsqueryoptions?view=azure-dotnet#feedback)
* Content Source: [xml/Azure.Monitor.Query/MetricsQueryOptions.xml](https://github.com/Azure/azure-docs-sdk-dotnet/blob/master/xml/Azure.Monitor.Query/MetricsQueryOptions.xml)
* Service: **azure**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

### PR DESCRIPTION
**Pull request by nisha-bhatia**
Fixes:  https://github.com/Azure/azure-sdk-for-net/issues/30903
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by pallavit — APPROVED** [review]
Ship it!

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 1]
```diff
@@ -546,6 +547,22 @@ foreach (MetricTimeSeriesElement element in metric.TimeSeries)
 }
 ```
 
+#### Get metrics namespaces
+
+To programmatically retrieve metrics namespaces, use the following code:
+    
+```C# Snippet:GetMetricsNamespaces
+string resourceId =
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Query/README.md:555]: The [issue](https://github.com/Azure/azure-sdk-for-net/issues/30903) that we are fixing seems to point to some confusion about how the `MetricNamespace` property of `MetricsQueryOptions` should be set. Could we add an example that shows the correct way to specify this property?

Comment (JoshLove-msft) [reply sdk/monitor/Azure.Monitor.Query/README.md:555]: Also, maybe we can update the ref docs for this property - https://learn.microsoft.com/en-us/dotnet/api/azure.monitor.query.metricsqueryoptions.metricnamespace?view=azure-dotnet#azure-monitor-query-metricsqueryoptions-metricnamespace

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.Query/README.md (hunk 1 above)
Comment (nisha-bhatia) [reply sdk/monitor/Azure.Monitor.Query/README.md:555]: I updated the parameter description to include the aka.ms link Scott created has a list of all the available MetricNamespaces - so hopefully that covers it?

Comment (JoshLove-msft) [reply sdk/monitor/Azure.Monitor.Query/README.md:555]: I see that we updated the ref doc for the property so that is probably good enough.

**Pull request merged**